agent/checks/check.go

```diff
@@ -577,7 +578,13 @@ func (c *CheckDocker) Start() {
 	}
 
 	if c.Logger == nil {
-		c.Logger = log.New(ioutil.Discard, "", 0)
+		consulLogger := hclog.New(&hclog.LoggerOptions{
```

Comment:
Maybe we could add a `testutil` helper like `NewDiscardLogger` that does this blob of code.